vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ImageService.java

```diff
@@ -250,4 +248,40 @@ public void removeCategory(IVSImage image, ImageCategory category) {
         image.getCategories().remove(category);
         imageRepo.save((VSImage) image);
     }
+    
+    /**
+     * Retrieves a list of IVSImages for pagination based on the provided parameters.
+     * 
+     * @param pageNo.    if pageNo<1, 1st page is returned, if pageNo>total
+     *                   pages,last page is returned
+     * @param searchTerm - This is the search string which is being searched.
+     * @param category - The category of the images to filter by.
+     * @return list of images in the requested pageNo and requested order.
+     */
+    @Override
+    public List<VSImage> getPaginatedImagesByCategoryAndSearchTerm(int pageNo, ImageCategory category, String searchTerm) {    	
+    	Sort sortingParameters = getSortingParameters(searchTerm, searchTerm);
+        pageNo = validatePageNumber(pageNo, category);
+        Pageable sortByRequestedField = PageRequest.of(pageNo - 1, pageSize, sortingParameters);
+        String likeSearchTerm = "%" + searchTerm + "%";
+        Page<VSImage> result = imageRepo.findByFilenameLikeOrNameLikeOrDescriptionLike(sortByRequestedField,
+            likeSearchTerm, likeSearchTerm, likeSearchTerm);
+        return result.getContent();
+    }
+      
+    /**
+     * Method to return the total pages sufficient to display all images
+     * with respect to the search text
+     * 
+     * @param searchTerm - This is the search string which is being searched.
+     * 
+     * @return totalPages required to display images with filename, name or description
```

Comment:
I don't understand this description. Isn't this simply the total pages for all search results?